### PR TITLE
use std::size_t and mp_integer in interpreter

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: src/goto-programs/interpreter.cpp [hunk 1]
```diff
@@ -679,9 +681,9 @@ void interpretert::execute_assign()
     side_effect_exprt side_effect=to_side_effect_expr(code_assign.rhs());
     if(side_effect.get_statement()==ID_nondet)
     {
-      unsigned address=integer2unsigned(evaluate_address(code_assign.lhs()));
-      size_t size=get_size(code_assign.lhs().type());
-      for(size_t i=0; i<size; i++)
+      std::size_t address=integer2unsigned(evaluate_address(code_assign.lhs()));
```
Comment (mgudemann) [inline src/goto-programs/interpreter.cpp]: should be `integer2size_t`

**Review by mgudemann — CHANGES_REQUESTED** [review]

File: src/goto-programs/interpreter.cpp (hunk 1 above)
Comment (mgudemann) [reply src/goto-programs/interpreter.cpp]: somehow related to this PR, there are also occurrences of `size_t` and not `std::size_t`

**Review by tautschnig — CHANGES_REQUESTED** [review]
Can this please be synchronised with #1330?

**Comment by kroening** [comment]
See comments in #1330.

**Comment by kroening** [comment]
There are a few remaining issues in the code, that could be addressed in a separate PR. In particular, we could worry about the case in which an address actually does exceed a 64-bit integer; or where it exceeds a size_t, and a vector is used. We could then give a graceful error message.

**Review by mgudemann — APPROVED** [review]
LGTM but the name of the PR should probably be changed now to include the `mp_integer` additions

**Review by reuk — APPROVED** [review]

**Comment by mgudemann** [comment]
@tautschnig are you ok with the comments on #1330 , i.e., using `mp_integer` here instead of any atomic integral type?

**Comment by mgudemann** [comment]
@tautschnig would you mind having a look, considering @kroening 's comment?

**Review by tautschnig — APPROVED** [review]
I am assuming that #1330 should just be closed as this PR effectively does the same?

**Comment by tautschnig** [comment]
> @tautschnig would you mind having a look, considering @kroening 's comment?

Been looking right in this moment ...

**Comment by peterschrammel** [comment]
@mgudemann, does this make https://github.com/diffblue/cbmc/pull/987 obsolete?

**Comment by mgudemann** [comment]
@peterschrammel yes, this is also an `mp_integer` now. The only thing is that it says `2^32` in the comment, but actually returns `2^33`
```
  if(unbounded_size(type))
    return mp_integer(2) << 32;
```

but the issue on 32-bit should be resolved with this PR

**Comment by peterschrammel** [comment]
@mgudemann, then please close https://github.com/diffblue/cbmc/pull/987.

**Pull request merged**